src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationView.Properties.cs

```diff
@@ -185,7 +185,7 @@ public DataTemplate HeaderTemplate
 		/// Identifies the HeaderTemplate dependency property.
 		/// </summary>
 		public static DependencyProperty HeaderTemplateProperty { get; } =
-			DependencyProperty.Register(nameof(HeaderTemplate), typeof(DataTemplate), typeof(NavigationView), new PropertyMetadata(null, OnPropertyChanged));
+			DependencyProperty.Register(nameof(HeaderTemplate), typeof(DataTemplate), typeof(NavigationView), new FrameworkPropertyMetadata(null, OnPropertyChanged));
```

Comment:
Be careful, DPs of type `DataTemplate` or `Style` should be marked `FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext`.

Reply:
Thanks for noting, I'll fix that.

Reply:
Fixed. Note that some existing properties in master were already using `FrameworkPropertyMetadata` but not specifying the option. Example from master:

https://github.com/unoplatform/uno/blob/890207ec33affe032e5251ddc2276704520ba63b/src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.Properties.cs#L175-L180

Reply:
Also note that this PR only handles `DependencyProperty.Register`, but not `RegisterAttached`. If both should be handled, please let me know whether you want it in the same PR or in a separate PR. Thanks!

Reply:
I think it should be the same for `RegisterAttached()`, but it probably makes sense to put it in a separate PR and close this one off.